packages/frontend/src/components/Game/GameBoard/index.test.jsx

```diff
@@ -117,4 +117,81 @@ describe('GameBoard component', () => {
     await user.click(tile.container('tile-7').get());
     expect(tile.container('tile-7').get()).not.toHaveClass('faceUp');
   });
+
+  describe('player turn', () => {
+    it('does not award a point for a non match', async () => {
+      const tiles = createTilesFromPhotos(mockPhotos, { shuffle: false });
+
+      const initialGameState = produce(baseState.game, (draft) => {
+        draft.tiles = tiles;
+      });
+
+      const state = { ...baseState, game: initialGameState };
+
+      const { user, screen } = setupTests(GameBoard, { state });
+
+      const currentPlayerScore = screen.getByTestId('player-score-1');
+
+      expect(currentPlayerScore).toHaveTextContent('0');
+
+      await user.click(tile.container('tile-5').get());
+      await user.click(tile.container('tile-2').get());
+      act(() => jest.advanceTimersByTime(3000));
+
+      const updatedScore = screen.getByTestId('player-score-1');
+      expect(updatedScore).toHaveTextContent('0');
+      screen.debug(updatedScore);
+    });
+
+    it('changes the active player after each turn', async () => {
+      const tiles = createTilesFromPhotos(mockPhotos, { shuffle: false });
+
+      const initialGameState = produce(baseState.game, (draft) => {
+        draft.tiles = tiles;
+      });
+
+      const state = { ...baseState, game: initialGameState };
+
+      const { user, screen } = setupTests(GameBoard, { state });
+
+      const activePlayerNote = screen.getByRole('note');
+      expect(activePlayerNote).toHaveTextContent('Player 1');
```

Comment:
Since this note is conditionally rendered, this query can be `expect(getByRole('note', { name: 'Player 1' } )).toBeInTheDocument()`. Adding the `name` option specifies which element with `role="note"` that it should find, so if there end up being more than one element with that role, RTL will know which one to find.

Reply:
The player indicator note can also be added to the `ui` object, much like the tiles, so the queries can be reused.

Reply:
The `ui` object might even be able to be exported from a single source so it could be reused in all the test files

Reply:
I couldn't get the test selector to work exactly like this, but what I have is close and seems to work. 

This way, it gets the span by role and then in the tests I check that the element has the correct text content. The other way was checking that the span and the text content together existed as an element in the document or not, and that was giving me problems.

Reply:
It's probably fine the way it is. The only problem we'd run into is if we added another element with the role of `note`, in which case RTL would find both and wouldn't know which one to pick. Providing the `name` property in the options object allows it to be more specific when selecting elements.

Can I ask what the error was that it was giving you? It seems like it should be possible to provide the name in the options object based on the [`testing-library-selector` docs](https://testing-library.com/docs/ecosystem-testing-library-selector/)
![image](https://github.com/perennialAutodidact/memory-snap/assets/56051123/0e575343-3ce7-4056-ba32-a554de313a6e)

Reply:
If it gives you an error about the name text being in multiple elements, the `name` property also accepts a (TextMatch)[https://testing-library.com/docs/queries/about#textmatch] function which can be used to specify the matching logic to be able to find the text. (TextMatch)[https://testing-library.com/docs/queries/about#textmatch]